worker/uniter/runner/jujuc/credential-get.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jujuc
+
+import (
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+)
+
+// CredentialGetCommand implements the leader-get command.
+type CredentialGetCommand struct {
+	cmd.CommandBase
+	ctx Context
+	out cmd.Output
+}
+
+// NewCredentialGetCommand returns a new CredentialGetCommand with the given context.
+func NewCredentialGetCommand(ctx Context) (cmd.Command, error) {
+	return &CredentialGetCommand{ctx: ctx}, nil
+}
+
+// Info is part of the cmd.Command interface.
+func (c *CredentialGetCommand) Info() *cmd.Info {
+	doc := `
+credential-get returns the cloud specification used by the unit's model.
+`
+	return &cmd.Info{
+		Name:    "credential-get",
+		Args:    "",
```

Comment:
I think at one point in the spec, we wanted to have the charm say something like:
credential-get cloud
vs
credential-get juju

Does that still seem relevant?
What would we name the command to get a Juju credential (something that lets you see/manipulate the juju model).

credential-get cloud

feels like something that lets us extend the functionality in the future, without being onerous to charms today (and means they won't have to be updated yet-again if we do start providing Juju credentials to continue getting cloud credentials in a clean fashion)

Reply:
I agree, yet the law of least work would dictate that we achieve any future backward compatibility by declaring `cloud` to be the default given there are not parameters past.

credential-get (returns cloud creds)
credential-get cloud (returns cloud creds)
credential-get juju (returns model creds)